Tests/MongoSwiftTests/ChangeStreamTests.swift

```diff
@@ -3,6 +3,274 @@ import mongoc
+
+    /// An array of additional aggregation pipeline stages to add to the change stream
+    let changeStreamPipeline: [Document]
+
+    /// Additional options to add to the changeStream
+    let changeStreamOptions: ChangeStreamOptions
+
+    /// Array of documents, each describing an operation.
+    let operations: [ChangeStreamTestOperation]
+
+    /// A list of command-started events in Extended JSON format.
+    let expectations: [Document]?
+
+    // The expected result of running a test.
+    let result: ChangeStreamTestResult
+
+    internal func run(globalClient: MongoClient, database: String, collection: String) throws {
+        let client = try MongoClient()
+
+        // TODO SWIFT-535: add APM assertions
+        do {
+            let changeStream = try self.target.watch(client,
+                                                     database,
+                                                     collection,
+                                                     self.changeStreamPipeline,
+                                                     self.changeStreamOptions)
+            for operation in self.operations {
+                _ = try operation.execute(using: globalClient)
+            }
+
+            switch self.result {
+            case .error:
+                _ = try changeStream.nextOrError()
+                fail("\(self.description) failed: expected error but got none while iterating")
+            case let .success(events):
+                var seenEvents: [ChangeStreamTestEvent] = []
+                for _ in 0..<events.count {
+                    let event = try changeStream.nextOrError()
+                    expect(event).toNot(beNil())
+                    seenEvents.append(event!)
+                }
+                expect(seenEvents).to(equal(events), description: self.description)
+            }
+        } catch {
+            self.result.matchesError(error: error, description: self.description)
+        }
+    }
+}
+
+/// Struct representing a single change-streams spec test JSON file.
+private struct ChangeStreamTestFile: Decodable {
+    private enum CodingKeys: String, CodingKey {
+        case databaseName = "database_name",
+             collectionName = "collection_name",
+             database2Name = "database2_name",
+             collection2Name = "collection2_name",
+             tests
+    }
+
+    /// Name of this test case
+    var name: String = ""
+
+    /// The default database.
+    let databaseName: String
+
+    /// The default collection.
+    let collectionName: String
+
+    /// Secondary database
+    let database2Name: String
+
+    // Secondary collection
+    let collection2Name: String
+
+    /// An array of tests that are to be run independently of each other.
+    let tests: [ChangeStreamTest]
+}
+
+final class ChangeStreamSpecTests: MongoSwiftTestCase, FailPointConfigured {
+    var activeFailPoint: FailPoint?
+
+    override func tearDown() {
+        self.disableActiveFailPoint()
```

Comment:
it might be helpful to clarify why we need this on top of the deferred one in a comment (I think the answer is that if the test fails the defer won't be reached...?)